### PR TITLE
Issue #10 make GetKey more flexible

### PR DESCRIPTION
**Pull request by jnardone**
I attempted to run the test suite (I'm not a regular .NET dev these days) and there's one failing test but I believe that's unrelated to this PR, but let me know.

### CODE REVIEW

File: Kerberos.NET/Entities/DecryptedData.cs [hunk 1]
```diff
@@ -109,7 +109,7 @@ protected virtual void ValidateTicketSkew(DateTimeOffset now, TimeSpan skew, Dat
 
         protected virtual void ValidateClientPrincipalIdentifier()
         {
-            if (!Ticket.CName.Equals(Authenticator.CName))
+            if (!Ticket.CName.Matches(Authenticator.CName))
```
Comment (jnardone) [inline Kerberos.NET/Entities/DecryptedData.cs]: I *think* this is right but not sure.

Comment (SteveSyfuhs) [reply Kerberos.NET/Entities/DecryptedData.cs]: That looks right. What's the failing unit test?

Comment (jnardone) [reply Kerberos.NET/Entities/DecryptedData.cs]: `testParseClaims` - I see from a previous PR that perhaps that's expected.

Comment (SteveSyfuhs) [reply Kerberos.NET/Entities/DecryptedData.cs]: Ah yeah. You're running this on not-Windows? I should eventually just add a platform check so the test behaves correctly.

Comment (jnardone) [reply Kerberos.NET/Entities/DecryptedData.cs]: Actually I did run this on Windows, but on a non-domain-connected box?  I get claims back in that test, but only 11 of them and not the one the last 
`Assert.IsTrue(result.Claims.Any(c=>c.Type == "ad://ext/employeeType:88d4d68c56082042" && c.Value == "lazy"));`

Comment (SteveSyfuhs) [reply Kerberos.NET/Entities/DecryptedData.cs]: Oh, interesting. This doesn't require being domain joined, but the compression algorithm for the claims may not be available. Are you on Win7 or Win8? What claims are present on the identity?

Comment (SteveSyfuhs) [reply Kerberos.NET/Entities/DecryptedData.cs]: Also, FWIW the compression algorithm is why claims don't work on non-Windows or earlier devices... I can't be bothered to build that particular algorithm. :)

Comment (jnardone) [reply Kerberos.NET/Entities/DecryptedData.cs]: This is Win7.  When I enumerate the claims in that test I get

```
result.Claims.ToList()
Count = 12
    [0]: {http://schemas.xmlsoap.org/ws/2005/05/identity/claims/sid: S-1-5-21-842315761-3748032240-3360761689-500}
    [1]: {http://schemas.xmlsoap.org/ws/2005/05/identity/claims/nameidentifier: Administrator@identityintervention.com}
    [2]: {http://schemas.microsoft.com/ws/2008/06/identity/claims/groupsid: S-1-5-21-842315761-3748032240-3360761689-512}
    [3]: {http://schemas.microsoft.com/ws/2008/06/identity/claims/role: Domain Admins}
    [4]: {http://schemas.microsoft.com/ws/2008/06/identity/claims/groupsid: S-1-5-21-842315761-3748032240-3360761689-513}
    [5]: {http://schemas.microsoft.com/ws/2008/06/identity/claims/role: Domain Users}
    [6]: {http://schemas.microsoft.com/ws/2008/06/identity/claims/groupsid: S-1-5-21-842315761-3748032240-3360761689-520}
    [7]: {http://schemas.microsoft.com/ws/2008/06/identity/claims/role: Group Policy Creator Owners}
    [8]: {http://schemas.microsoft.com/ws/2008/06/identity/claims/groupsid: S-1-5-21-842315761-3748032240-3360761689-519}
    [9]: {http://schemas.microsoft.com/ws/2008/06/identity/claims/role: Enterprise Admins}
    [10]: {http://schemas.microsoft.com/ws/2008/06/identity/claims/groupsid: S-1-5-21-842315761-3748032240-3360761689-518}
    [11]: {http://schemas.microsoft.com/ws/2008/06/identity/claims/role: Schema Administrators}
```

Comment (SteveSyfuhs) [reply Kerberos.NET/Entities/DecryptedData.cs]: Okay, that's expected. All looks good. Approving.

**Pull request merged**